app/mailers/app_mailer.rb

```diff
@@ -2,21 +2,22 @@ class AppMailer < ActionMailer::Base
 
   default from: 'admin@myflix.com'
 
-  def send_mail_on_register(user)
-    @user = user
-    mail to: user.email, subject: 'You have registered successfully!'
+  def send_mail_on_register(user_id)
+    @user = User.find(user_id)
+    mail to: @user.email, subject: 'You have registered successfully!'
   end
 
-  def send_mail_on_reset_password(user)
-    @user = user
-    @link = reset_password_url(user.reset_password_token, Rails.application.config.action_mailer.default_url_options)
-    mail to: user.email, subject: 'Reset your password now'
+  def send_mail_on_reset_password(user_id)
+    @user = User.find(user_id)
+    @link = reset_password_url(@user.reset_password_token, Rails.application.config.action_mailer.default_url_options)
+    mail to: @user.email, subject: 'Reset your password now'
```

Comment:
Its great to have a filter method on who you send the mail based on your environment. TEST and DEVELOPMENT does not matter but for STAGING you could filter it to `admin@jepetko.at` maybe - something like that.